radicle-cli/src/commands/patch/checkout.rs

```diff
@@ -0,0 +1,39 @@
+use crate::terminal as term;
+use anyhow::anyhow;
+use radicle::cob::patch::{self, PatchId};
+use radicle::git::{self, RefStr};
+use radicle::profile::Profile;
+use radicle::storage::git::Repository;
+
+pub fn run(
+    storage: &Repository,
+    profile: &Profile,
+    git_workdir: &git::raw::Repository,
+    patch_id: &PatchId,
+) -> anyhow::Result<()> {
+    let patches = patch::Patches::open(profile.public_key, storage)?;
+    let patch = patches
+        .get(patch_id)?
+        .ok_or_else(|| anyhow!("Patch `{patch_id}` not found"))?;
+
+    let spinner = term::spinner("Performing patch checkout...");
+
+    // Getting the patch obj!
+    let patch_head = patch.head();
+    let obj = git_workdir.revparse_single(&patch_head.to_string())?;
+
+    let def_name = format!("patch/{}", term::format::cob(patch_id));
+    // checkout the patch in a new branch!
+    let commit = git_workdir.find_commit(obj.id())?;
+    git_workdir.branch(&def_name, &commit, false)?;
+    // and then point the current `HEAD` inside the new branch.
+    let branch_name = RefStr::try_from_str(&def_name)?;
+    let branch_name = git::refs::workdir::branch(branch_name);
+    git_workdir.set_head(&branch_name)?;
```

Comment:
I believe we can define the `name` for the branch all in one go here and then do the `branch` and `set_head`. IIRC `branch` will DWIM (Do What I Mean) and ignore the `refs/heads` for you, but maybe you can confirm that on your end.
```suggestion
    let name = RefStr::try_from_str(&format!("patch/{}", term::format::cob(patch_id)))?;
    let name = git::refs::workdir::branch(name);
    // checkout the patch in a new branch!
    let commit = git_workdir.find_commit(obj.id())?;
    git_workdir.branch(name.as_str(), &commit, false)?;
    // and then point the current `HEAD` inside the new branch.
    git_workdir.set_head(name.as_str())?;
```

Reply:
Yeah, but the question that I was not able to answer from the docs is if the git2 is able to strip the ref prefix? or just append it?

Also, I think that prin the ref prefix in the terminal is confusing for the normal user?

Reply:
Yeah, we don't want to print that to the user at least.

Reply:
So the suggestion will translate to the actual diff if I understand correctly the suggestion

Reply:
Ah, that's fair. You can change my suggestion to have the second `name` be `branch` instead then. So:
```suggestion
    let name = RefStr::try_from_str(&format!("patch/{}", term::format::cob(patch_id)))?;
    let branch = git::refs::workdir::branch(name);
    // checkout the patch in a new branch!
    let commit = git_workdir.find_commit(obj.id())?;
    git_workdir.branch(branch.as_str(), &commit, false)?;
    // and then point the current `HEAD` inside the new branch.
    git_workdir.set_head(branch.as_str())?;
```

Reply:
Mh not sure that this `    let name = RefStr::try_from_str(&format!("patch/{}", term::format::cob(patch_id)))?;` will work? but I will be back to it and think more deep

Reply:
You're right, it should be:
```rust
    let name = RefString::try_from(format!("patch/{}", term::format::cob(patch_id)))?;
    let branch = git::refs::workdir::branch(&name);
```

Reply:
Thanks!